ophyd/areadetector/cam.py

```diff
@@ -1289,6 +1290,171 @@ class RoperDetectorCam(CamBase):
     roper_shutter_mode = ADCpt(SignalWithRBV, "RoperShutterMode")
 
 
+class TimePix3Chip(Device):
+
+    def __init__(self, chip_number):
```

Comment:
The components need to be at the class level.  I think you want to use `FormattedComponents` https://github.com/bluesky/ophyd/blob/75a35fdee07f8e1ff9d865b31baae61636cd7ceb/ophyd/device.py#L369-L396 or `DynamicDeviceComponent` https://github.com/bluesky/ophyd/blob/75a35fdee07f8e1ff9d865b31baae61636cd7ceb/ophyd/device.py#L407-L437 here.